src/java/com/puppetlabs/ssl_utils/ExtensionsUtils.java

```diff
@@ -780,21 +801,63 @@ private static AuthorityKeyIdentifier mapToAuthorityKeyIdentifier(
                         "'serial_number' not provided for auth key identifier" +
                         "but was expected since 'issuer' was provided");
             }
+            return true;
+        }
+        return false;
+    }
 
-            GeneralNames issuerAsGeneralNames =
-                    new GeneralNames(new GeneralName(new X500Name(issuer)));
-            if (authorityKeyId != null) {
-                authorityKeyId = new AuthorityKeyIdentifier(
-                        authorityKeyId.getKeyIdentifier(),
-                        issuerAsGeneralNames,
-                        serialNumber);
-            }
-            else {
-                authorityKeyId = new AuthorityKeyIdentifier(
-                        issuerAsGeneralNames,
-                        serialNumber);
+    private static AuthorityKeyIdentifier authKeyIdFromIssuer(String issuer,
+                                                              AuthorityKeyIdentifier authorityKeyId,
+                                                              BigInteger serialNumber)
+        throws OperatorCreationException {
+        GeneralNames issuerAsGeneralNames =
+            new GeneralNames(new GeneralName(new X500Name(issuer)));
+        if (authorityKeyId != null) {
+            return new AuthorityKeyIdentifier(authorityKeyId.getKeyIdentifier(),
+                                                        issuerAsGeneralNames,
+                                                        serialNumber);
+        }
+        else {
+            return new AuthorityKeyIdentifier(issuerAsGeneralNames,
+                                                        serialNumber);
+        }
+    }
+
+    private static AuthorityKeyIdentifier mapToAuthorityKeyIdentifier(Map<String, Object> authKeyIdMap,
+                                                                      Boolean truncate)
+        throws OperatorCreationException, CertificateEncodingException, IOException {
+        AuthorityKeyIdentifier authorityKeyId = null;
+
+        PublicKey pubKey = (PublicKey) authKeyIdMap.get ("public_key");
+        if (pubKey != null) {
+            authorityKeyId = authKeyIdFromPubKey(pubKey, truncate);
+        }
+
+        BigInteger serialNumber = (BigInteger) authKeyIdMap.get
+                ("serial_number");
+        X509Certificate cert = (X509Certificate) authKeyIdMap.get("cert");
```

Comment:
Everything except for this line and the `authorityKeyId = null` that's top level could be nested inside the `cert == null` branch, right?